### PR TITLE
fix: perf regression caused by optimizeTree hook

### PR DESCRIPTION
**Pull request by ahabhgk**
## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by ahabhgk** [comment]
!bench

**Comment by ahabhgk** [comment]
!bench

**Review by h-a-n-a — APPROVED** [review]

File: crates/node_binding/src/hook.rs [hunk 1]
```diff
@@ -32,7 +32,7 @@ pub enum Hook {
   AfterCompile,
   FinishModules,
   OptimizeModules,
-  OptimizaTree,
+  OptimizeTree,
```
Comment (hardfist) [inline crates/node_binding/src/hook.rs:35]: don't we have typo check now?

Comment (ahabhgk) [reply crates/node_binding/src/hook.rs:35]: `optimiza tree` can be detected but `OptimizaTree` can't, I guess

**Pull request merged**